### ISSUE TITLE
Double Compile is Needed For Hot Reload Bug?

### ISSUE THREAD

**Issue by chobo2** [webpack/webpack#6996]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

Hi

I am using this template: https://github.com/bradtraversy/react_webpack_starter and I noticed something very strange.  When I load it up for the first time and change some html around it works just fine and I see my web browser page change.

Now when I try to make a new change nothing updates anymore. I have to compile it twice(ie hit the save button twice) for the hot reload to work again.

**If the current behavior is a bug, please provide the steps to reproduce.**

**What is the expected behavior?**

To reload on first save, not second save.

**If this is a feature request, what is motivation or use case for changing the behavior?**

**Please mention other relevant information such as the browser version, Node.js version, webpack version, and Operating System.**

I am getting this on Chrome(latest), node 8.10.0, webpack 4.5.0, windows 7

**Comment by evilebottnawi** [comment]
@chobo2 problem still exists? If yes please create minimum reproducible test repo. I can't reproduce using https://github.com/bradtraversy/react_webpack_starter

**Comment by chobo2** [comment]
@evilebottnawi 

Hey, I will have to let you know in a few weeks, have not been doing webpack right now(been doing backend code).

**Comment by chobo2** [comment]
I still am getting it, it seems like it still hangs.  I am using 

    "webpack": "^4.10.2",
    "webpack-cli": "^2.1.4",
    "webpack-dev-server": "^3.1.4"

I am not sure what else to give you for info.

**Comment by chobo2** [comment]
@evilebottnawi 

Here is a project I made, I tried to make it as similar as possible to my real code, though most of the packages I put in are not used.

https://github.com/chobo2/WebpackTest

So what I do is I got to app.js and I change the content div, so I will go

1. npm start
2. change div text to something other random text
3. watch chrome (it may update or it may not)
4. Will keep doing it till ti fails, usually 1-3 times of doing it will be enough.
